topydo/ui/columns/Main.py

```diff
@@ -43,6 +44,60 @@
 from topydo.ui.columns.ColumnLayout import columns
 
 
+class Transaction(object):
+    """
+    This class implements basic handling of executing any subcommand on multiple
+    todo items.
+    """
+    def __init__(self, p_todo_ids=None):
+        self._subcommand = None
+        self._todo_ids = p_todo_ids
+        self._operations = []
+
+    def prepare(self, p_subcommand, p_args):
+        """
+        Prepares list of operations to execute based on and p_args and list of
+        todo items contained in _todo_ids attribute. Also sets _subcommand
+        attribute from p_subcommand.
+        """
+        self._subcommand = p_subcommand
+
+        if self._todo_ids:
+            id_position = p_args.index('{}')
+
+            if issubclass(self._subcommand, MultiCommand):
+                p_args[id_position:id_position] = self._todo_ids
+                p_args.remove('{}')
+                self._operations.append(p_args)
+            else:
+                for todo_id in self._todo_ids:
+                    operation_args = p_args[:]
+                    operation_args.insert(id_position, todo_id)
+                    operation_args.remove('{}')
+                    self._operations.append(operation_args)
+        else:
+            self._operations.append(p_args)
+
+    def execute(self, p_todolist, p_out, p_err, p_input):
```

Comment:
We could reduce it to a single parameter, making it a bit more maintenance-friendly. In `Main.py`, when creating an `Transaction`, we define the environment first with a lambda:

```python
command_placeholder = lambda op: subcommand(op, p_output, self._output, self._input)
```

All that needs to be filled are the arguments per operation.

We could pass this lambda to the constructor of `Transaction`. This still allows you to check the type of the command for `MultiCommand` in `prepare()`, and easily execute the operation in `execute()` with (`self.command(operation)`).

That would also address the issue above that the subcommand should not be passed for every `prepare()`.